src/pyscipopt/scip.pyx

```diff
@@ -766,6 +780,16 @@ cdef class Model:
         """returns current limit on objective function."""
         return SCIPgetObjlimit(self._scip)
 
+    def delObjective(self):
+        """clear existing objective function"""
+        cdef SCIP_VAR** _vars
+        cdef int _nvars
+
+        _vars = SCIPgetOrigVars(self._scip)
+        _nvars = SCIPgetNOrigVars(self._scip)
+        for i in range(_nvars):
+            PY_SCIP_CALL(SCIPchgVarObj(self._scip, _vars[i], 0.0))
+
```

Comment:
Why is this necessary? Can't you just call `setObjective(0)`?

Reply:
setObjective(0) raises an AssertionError at  assert isinstance(coeffs, Expr) in setObjective. Therefore I created this method. Maybe 0 should be an instance of Expr, I don't know

Reply:
Oh! I was not aware of that. Then we should rather fix this instead of introducing a new method.

Reply:
This has been fixed now in the master branch. Please remove `delObjective()`.